src/main/java/com/ioet/bpm/peopletime/timeevents/services/LastActiveEventService.java

```diff
@@ -0,0 +1,26 @@
+package com.ioet.bpm.peopletime.timeevents.services;
+
+import com.ioet.bpm.peopletime.timeevents.domain.TimeEvent;
+import com.ioet.bpm.peopletime.timeevents.repositories.TimeEventRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import java.util.ArrayList;
+
+@AllArgsConstructor
+@Service
+public class LastActiveEventService {
+    private final TimeEventRepository timeEventRepository;
+
+    public Iterable<?> getLastActiveTimeEvents(String orderByCriteria, String personId, int top) {
+        if (orderByCriteria == null) {
+            return this.timeEventRepository.findByPersonId(personId);
+        } else if ("lastActive".equals(orderByCriteria)) {
+            ArrayList<TimeEvent> lastActiveTimeEvent = new ArrayList<>();
+            lastActiveTimeEvent.add(this.timeEventRepository.lastActiveTimeEvent(personId, top).get());
```

Comment:
If you pass a nonExisting personId, this will cause an exception, right?
You should first check if the optional has or not a value:
Optional<> value = this.timeEventRepository.lastActiveTimeEvent(personId, top)
if(value.exist())